### PR TITLE
Allow for using a dedicated binary for the torch subproc pool.

### PR DESCRIPTION
**Pull request by c00w**
Summary:
The binary torch is running inside of can be larger than needed and in certain
situations, this can cause a loss of memory.

Test Plan:
We've manually run tests via
```
TORCHINDUCTOR_FORCE_DISABLE_CACHES=1 TORCHINDUCTOR_WORKER_SUPPRESS_LOGGING=0
make mc8-train-publish-cint-datafm-toy -C
minimal_viable_ai/models/ifr_mtml/main_v1/ 2>&1 | tee ~/run_out
```
and overriding the binary used to be the built fbpkg in /packages.

We've also kicked off manual runs at
```
fire-feid-20250903-1051-ae8c6827
```

Which do show the binary running -  https://fburl.com/scuba/procprint/e6lwv32m

Rollback Plan:
steps:
  - jk.update:
      jk: pytorch/compiler:subproc_worker_binary
      constant_bool: null
      consistent_pass_rate: null
      fractional_host_rollout: null
      sampling_rate: null
  - manual.note:
      content: ''

Differential Revision: D81616624

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

### CODE REVIEW

File: torch/_inductor/config.py [hunk 1]
```diff
@@ -775,6 +775,10 @@ def decide_worker_start_method() -> str:
     default=True,
 )
 
+# Supply an override for a standalone compiler worker binary.
```
Comment (masnesral) [inline torch/_inductor/config.py]: Let's just remove this. Not sure how this got in anyway, probably an artifact of us both trying to work on each others' diffs at the same time

**Review by masnesral — CHANGES_REQUESTED** [review]

**Review by masnesral — APPROVED** [review]
